src/main/java/bio/terra/service/auth/iam/IamProviderInterface.java

```diff
@@ -125,11 +125,15 @@ Map<IamRole, String> createDatasetResource(
    *
    * @param userReq authenticated user
    * @param snapshotId id of the snapshot
+   * @param parentId id of the parent dataset of the snapshot
    * @param policies user emails to add as snapshot policy members
    * @return Map of policy group emails for the snapshot policies
    */
   Map<IamRole, String> createSnapshotResource(
-      AuthenticatedUserRequest userReq, UUID snapshotId, SnapshotRequestModelPolicies policies)
+      AuthenticatedUserRequest userReq,
+      UUID snapshotId,
+      UUID parentId,
```

Comment:
```suggestion
      UUID parentDatasetId,
```

For consistency with the rest of the code, I think it makes sense to rename this. Ditto with the other methods.

Reply:
OK. My original thinking was that the API didn't need to know that this was a dataset ID below the service layer, but the type ended up needing to be hardcoded in `createSnapshotResourceRequest()`.